### PR TITLE
Fix wait/wait_not timings, use time.clock(), extend list view detection

### PR DESCRIPTION
**Pull request by vasily-v-ryabov**
* `wait / wait_not` timings are handled properly now (passing params to `_resolve_control`) fixing #347.
* `time.clock()` is resistant to accident system clock change (I found it at the main job recently).
* Extended list view detection list (thanks @KirillMoizik ).

### CODE REVIEW

**Comment by airelil** [comment]
@cetygamer is right, it's hard to see the changes in application.py and common_controls.py

**Comment by vasily-v-ryabov** [comment]
It looks like someone of master students used non-default `autoCrLf` value in git config. I'll try to fix it later.
Also I've sent a feedback about GitHub diff viewer that could be improved (`git diff` is not the best option in my mind). Support team replied they will consider it probably.
Possible workaround for now is reviewing every commit except merges.

File: pywinauto/application.py [hunk 1]
```diff
@@ -1,1324 +1,1331 @@
...
+
+        If we already have criteria for both dialog and control then
+        resolve the control and return the requested attribute.
+
+        If we have only criteria for the dialog but the attribute
+        requested is an attribute of DialogWrapper then resolve the
+        dialog and return the requested attribute.
+
+        Otherwise delegate functionality to :func:`__getitem__` - which
+        sets the appropria
```
Comment (airelil) [inline pywinauto/application.py]: A good spot on the explicit call to ```self.__resolve_control``` and timings arguments! I have to admit, I don't understand why ```exists``` needs a special treatment, but if it's the case the ```exists``` processing should be moved a bit upper, before entering into the ```for``` loop, like this: 
```python 
if 'exists' in check_names:
    check = getattr(self, check_name)
    return check(retry_interval, float(retry_interval) // 2)

for check_name in check_names:
    try:
        ...
``` 
The current implementation will demonstrate unstable behavior if ```exists``` will be coupled with any other check.

File: pywinauto/application.py [hunk 2]
```diff
@@ -1044,10 +1044,10 @@ def wait_cpu_usage_lower(self, threshold=2.5, timeout=None, usage_interval=None)
         if timeout is None:
             timeout = Timings.cpu_usage_wait_timeout
 
-        start_time = time.time()
+        start_time = time.clock()
```
Comment (cetygamer) [inline pywinauto/application.py]: Alternatively we could use `timeit` module. It is supposed to be more reliable in terms of calculating elapsed time on different operating systems. See https://docs.python.org/2/library/timeit.html#timeit.default_timer

Comment (airelil) [inline pywinauto/application.py]: I think, ```timeit``` is used mostly for code profiling.

Comment (airelil) [inline pywinauto/application.py]: @vasily-v-ryabov, we may make these time.clock changes later or in a separate PR

File: pywinauto/application.py (hunk 1 above)
Comment (vasily-v-ryabov) [inline pywinauto/application.py]: @airelil you're partially right. It made me think to write one more unit test to catch the logic fail for case `'exists visible enabled'`. Not sure I will fix it suggested way. But I definitely shouldn't return immediately if `exists` check is True.

File: pywinauto/application.py (hunk 2 above)
Comment (vasily-v-ryabov) [inline pywinauto/application.py]: OK, will make separate pull requests.
